### ISSUE TITLE
TokenExpired right after auth

### ISSUE THREAD

**Issue by magobaol** [ledesmablt/spotify-cli#6]
I've just installed this and right after being succesfully authenticated with Spotify, for any command I try to execute I receive this message

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/utils/Spotify.py", line 104, in _handle_request
    with urlopen(req) as res:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 401: Unauthorized

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/utils/Spotify.py", line 152, in request
    res_data = _handle_request(endpoint, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/utils/Spotify.py", line 130, in _handle_request
    raise TokenExpired
cli.utils.exceptions.TokenExpired: <exception str() failed>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/utils/Spotify.py", line 104, in _handle_request
    with urlopen(req) as res:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 401: Unauthorized

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/commands/devices.py", line 26, in devices
    res = Spotify.request('me/player/devices', method='GET')
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/utils/Spotify.py", line 157, in request
    res_data = _handle_request(endpoint, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cli/utils/Spotify.py", line 130, in _handle_request
    raise TokenExpired
cli.utils.exceptions.TokenExpired: <exception str() failed>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/bin/spotify", line 8, in <module>
    sys.exit(cli())
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 799, in main
    e.show()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/exceptions.py", line 42, in show
    echo("Error: {}".format(self.format_message()), file=file)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/exceptions.py", line 28, in format_message
    return self.message
AttributeError: 'TokenExpired' object has no attribute 'message'
```

Can you help?

**Comment by ledesmablt** [comment]
Hi @magobaol could you try running `spotify auth status -v`? It should show your name and the location of the credentials in case you're logged in. Also what OS are you using?

Tried a clean install somewhere else and it works for me on both windows and linux, but I'll look into the issue. Will also give it a prettier error message. Thanks!

**Comment by magobaol** [comment]
Hi, that command correctly shows my Spotify username and the location

```
Logged in as Francesco Face
Credentials stored in /Users/francesco/.config/spotify-cli/credentials.json
```

I'm using macOS 10.14

**Comment by benjshopee** [comment]
Hmm since your name shows up, that means the refresh token is working for that API endpoint (`GET /me`). Every request to the Spotify API uses the `access_token` and `refresh_token` values from the credentials file. This one succeeded, so all other requests should be working.

Could you try running this python command to check the creds file? Should be similar to the expected output below. Also make sure that you're on the latest version of spotify-cli - 0.3.3. Thanks!
```bash
# command
python3 -c "path='/Users/francesco/.config/spotify-cli/credentials.json';import json;f=open(path, 'r');d=json.load(f);print(d.keys())"

# expected output
dict_keys(['access_token', 'expires_in', 'refresh_token', 'scope', 'token_type', 'auth_code'])
```

**Comment by ledesmablt** [comment]
Hmm since your name shows up, that means the refresh token is working for that API endpoint (`GET /me`). Every request to the Spotify API uses the `access_token` and `refresh_token` values from the credentials file. This one succeeded, so all other requests should be working.

Could you try running this python command to check the creds file? Should be similar to the expected output below. Also make sure that you're on the latest version of spotify-cli - 0.3.3. Thanks!
```bash
# command
python3 -c "path='/Users/francesco/.config/spotify-cli/credentials.json';import json;f=open(path, 'r');d=json.load(f);print(d.keys())"

# expected output
dict_keys(['access_token', 'expires_in', 'refresh_token', 'scope', 'token_type', 'auth_code'])
```

**Comment by magobaol** [comment]
Hi,
I confirm that the output is precisely as expected, and that the version is 0.3.3.
I see what you mean, that's weird, indeed.

**Comment by ledesmablt** [comment]
Could you run `spotify auth login` again and try other commands to see if they're working? Doubt this will work, so I'll test this on macOS somehow and will keep this thread posted.

**Comment by magobaol** [comment]
Tried, it didn't work. Also tried to delete both files in `~/.config/spotify-cli` and re-auth. Auth was fine but no luck with the other commands.

**Comment by magobaol** [comment]
If you don't have a macOS at hand, let me know if I can help you in some way.

**Comment by ledesmablt** [comment]
Thank you for offering. One last thing, is your account on Spotify premium? I think some features are inaccessible to non-premium users, but I wasn't able to test it myself.

If you'd like to debug it yourself, I suggest you clone the repo and run `python3 main.py` as the cli entrypoint (instead of just `spotify`). You could print the headers in the `cli.utils.Spotify._handle_request` function (line 81-138) to see if the auth token is being sent to the API.

The authorization error should be related to having an invalid auth token, but the CLI is already supposed to handle token refreshes in that case (given that the auth code you copied from the `spotify auth login` flow is correct). I'll still try to figure this out on my end, in case there are other unknown compatibility issues.

Many thanks man!

**Comment by magobaol** [comment]
I have a Spotify Premium account.

I've printed out the headers for both calls (auth and devices) and they are identical.

Then I tried another thing: I obtained a Bearer token through the Spotify Developer Console and put that one inside the `credentials.json`, replacing the one obtained through `auth login`. 

Made the call to `devices` and it worked. So it seems something tied to the oauth scope, but I can't figure out what, since it seems to me that you are using the right scopes.

Let me know if I can do anything else
